app/src/main/java/ch/epfl/sweng/SDP/game/RankingFragment.java

```diff
@@ -78,9 +111,37 @@ public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
                     }
                 }
 
-                rankedUsernames = SortUtils.sortByValue(finalRanking);
+                String accountId = Account.getInstance(getActivity().getApplicationContext()).getUserId();
+                String userNameId = Account.getInstance(getActivity().getApplicationContext()).getUsername();
+
+                // Sort the rankings
+                List<String> rankingUsernames = SortUtils.sortByValue(finalRanking);
+                Integer[] rankings = (Integer[])(finalRanking.values().toArray(new Integer[finalRanking.values().size()]));
+                Arrays.sort(rankings, Collections.reverseOrder());
+
+                int rankingForUser = dataSnapshot.child(userNameId).getValue(Integer.class);
+
+                // Calculate trophies
+                Integer[] trophies = new Integer[rankings.length];
+                int rank = 10;
```

Comment:
It might be better to create a final static value RANK at the beginning of this class?